arraycontainer.go

```diff
@@ -359,35 +359,24 @@ func (ac *arrayContainer) iorArray(value2 *arrayContainer) container {
 	len1 := value1.getCardinality()
 	len2 := value2.getCardinality()
 	maxPossibleCardinality := len1 + len2
-	if maxPossibleCardinality > arrayDefaultMaxSize { // it could be a bitmap!
-		bc := newBitmapContainer()
-		for k := 0; k < len(value2.content); k++ {
-			v := value2.content[k]
-			i := uint(v) >> 6
-			mask := uint64(1) << (v % 64)
-			bc.bitmap[i] |= mask
-		}
-		for k := 0; k < len(ac.content); k++ {
-			v := ac.content[k]
-			i := uint(v) >> 6
-			mask := uint64(1) << (v % 64)
-			bc.bitmap[i] |= mask
-		}
-		bc.cardinality = int(popcntSlice(bc.bitmap))
-		if bc.cardinality <= arrayDefaultMaxSize {
-			return bc.toArrayContainer()
-		}
-		return bc
-	}
 	if maxPossibleCardinality > cap(value1.content) {
-		newcontent := make([]uint16, 0, maxPossibleCardinality)
+		// doubling the capacity reduces new slice allocations in the case of
+		// repeated calls to iorArray().
+		newcontent := make([]uint16, 0, 2*maxPossibleCardinality)
```

Comment:
I am mildly preoccupied by this change. If maxPossibleCardinality is 2* arrayDefaultMaxSize, then you will allocate 4* arrayDefaultMaxSize which is twice as much as you will ever need. So it seems like you might overallocate, maybe quite often.

That may not translate into visibly poorer performance, or even into higher memory usage, in a given benchmark, but given that you are clearly motivated here by mem allocs, it seems like we would like the narrow the bound.

So doubling is maybe fine, but then if it exceeds 2* arrayDefaultMaxSize, just set it to 2* arrayDefaultMaxSize. Or something.


Thoughts?